File: Java Food Order App/RegularCustomer.java
package FoodOrderApp;

public class RegularCustomer extends Customer implements Bill {
	
		float discount = 6.5f;
		int deliveryCharges = -50;
		static String type = "Regular Customer";
		
		RegularCustomer(byte customerID, Name customerName, long customerPhone, Address customerAddress, String customerEmail) {
			super(customerID, customerName, customerPhone, customerAddress, customerEmail);
		}
		
		public void payBill(float total) {	
			System.out.println("\tTotal Price is: " + total);
			System.out.println("\n\tSince you are our " + RegularCustomer.type);
			float billAmount = total - (total * discount / 100) + deliveryCharges;
			System.out.println("\tAfter Discount: " + billAmount);
		}
}

File: Kotlin Food Order App/RegularCustomer.kt
package foodOrderApp

class RegularCustomer (
    customerID : Byte,
    customerName: Name,
    customerPhone: Long,
    customerAddress: Address,
    customerEmail: String
) : Customer(customerID.toByte(), customerName, customerPhone, customerAddress, customerEmail), BillOrder {

    private var discount: Float = 6.5f
    private var deliveryCharges: Int = -50

    companion object {
        var type = "Regular Customer"
    }

    override fun payBill(total: Float) {
        println("\tTotal Price is: $total")
        println("\n\tSince you are our $type")
        val billAmount: Float = (total - (total * discount / 100)) + deliveryCharges
        println("\tAfter Discount: $billAmount")
    }

}